src/librustc_hir/lib.rs

```diff
@@ -3,7 +3,9 @@
 //! [rustc guide]: https://rust-lang.github.io/rustc-guide/hir.html
 
 #![feature(crate_visibility_modifier)]
+#![feature(const_if_match)]
```

Comment:
you could all `#[allow_internal_unstable(const_if_match, const_panic)]` to the macro so that the macro-using crates don't need all these feature gates

Reply:
I tried it just now on the macro, the method definitions and both. It seems `newtype_index` defines new functions and `allow_internal_unstable` doesn't apply transitively, so it doesn't work.

Reply:
oh... curious. well, this is just rust internal stuff, so ok